### PR TITLE
feat: Github connector Connector settings - MEED-2195 - Meeds-io/MIPs#64

### PR DESCRIPTION
**Pull request by AzmiTouil**
This PR will make sure to validate users' GitHub accounts before saving using OauthService

### CODE REVIEW

File: gamification-github-services/src/main/java/org/exoplatform/gamification/github/services/impl/GithubAccountServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package org.exoplatform.gamification.github.services.impl;
```
Comment (boubaker) [inline gamification-github-services/src/main/java/org/exoplatform/gamification/github/services/impl/GithubAccountServiceImpl.java:16]: missing license header

File: gamification-github-webapp/src/main/webapp/js/githubConnector.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  isSignedIn: true,
+  identifier: '',
+  user: '',
+  rank: 10,
+  openOauthPopup(connector) {
+    const width = 500;
+    const height = 600;
+    const left = window.innerWidth / 2 - width / 2;
+    const top = window.innerHeight / 2 - height / 2;
+    const authUrl = `https://github.com/login/oauth/authorize?client_id=${connector.apiKey}&redirect_uri=${connector.redirectURL}`;
```
Comment (boubaker) [inline gamification-github-webapp/src/main/webapp/js/githubConnector.js]: `redirectURL` must be a constant provided from here depending on used domain and not provided in connector settings.

```suggestion
    const authUrl = `https://github.com/login/oauth/authorize?client_id=${connector.apiKey}&redirect_uri=${redirectURL}`;
```

File: gamification-github-services/src/main/resources/conf/portal/configuration.xml [hunk 3]
```diff
@@ -54,6 +54,29 @@
...
+          <value>${meeds.gamification.connector.github.key:}</value>
+        </value-param>
+        <value-param>
+          <name>connectorSecretKey</name>
+          <value>${meeds.gamification.connector.github.secret:}</value>
+        </value-param>
+        <value-param>
+          <name>connectorRedirectURL</name>
+          <value>${meeds.gamification.connector.github.redirectURL:}</value>
+        </value-param>
```
Comment (boubaker) [inline gamification-github-services/src/main/resources/conf/portal/configuration.xml]: This parameter should be built in UI depending on used domain

```suggestion
```

File: gamification-github-services/pom.xml [hunk 4]
```diff
@@ -106,6 +106,22 @@
...
+        <dependency>
+            <groupId>org.exoplatform.social</groupId>
+            <artifactId>social-component-oauth-auth</artifactId>
+            <version>6.5.x-meedsv2-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.exoplatform.addons.gamification</groupId>
+            <artifactId>gamification-services</artifactId>
+            <version>2.5.x-meedsv2-SNAPSHOT</version>
```
Comment (boubaker) [inline gamification-github-services/pom.xml]: Shouldn't set version

```suggestion
```

File: gamification-github-services/pom.xml [hunk 5]
```diff
@@ -106,6 +106,22 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.github.scribejava</groupId>
+            <artifactId>scribejava-apis</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.exoplatform.social</groupId>
+            <artifactId>social-component-oauth-auth</artifactId>
+            <version>6.5.x-meedsv2-SNAPSHOT</version>
```
Comment (boubaker) [inline gamification-github-services/pom.xml]: Shouldn't set version here, use parent POM to add thrid party dependencies version as made in other addons

```suggestion
```

File: gamification-github-services/pom.xml [hunk 6]
```diff
@@ -106,6 +106,22 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.exoplatform.addons.gamification</groupId>
+            <artifactId>gamification-services</artifactId>
+            <version>2.5.x-meedsv2-SNAPSHOT</version>
+            <scope>compile</scope>
```
Comment (boubaker) [inline gamification-github-services/pom.xml]: Why using the scope `compile`, it should be `provided` for all JARs that will not be packaged inside the addon ZIP.

```suggestion
```

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: gamification-github-services/pom.xml (hunk 6 above)
Comment (AzmiTouil) [reply gamification-github-services/pom.xml]: Done!

**Pull request merged**